### DISCUSSION TITLE
Handling Deep Links on Android

### DISCUSSION THREAD

**Discussion by csbence10** [Milad-Akarie/auto_route_library#2112] [Q&A]
I am facing an issue with Android App Links (deep linking) when using auto_route. The setup for Android App Links is correct, and the app is successfully opened from a terminated state when clicking on a link like https://mydomain.com/link/feed. However, the app throws the following error upon launch: `Cannot resolve initial route`

I believe the issue lies on the Flutter side or in how `auto_route` handles initial routes when the app is launched via a deep link.

**Defined Routes**

These are the routes configured in my app:

```
@override
List<AutoRoute> get routes => [
  AutoRoute(
    initial: true,
    path: '/login',
    page: LoginRoute.page,
  ),
  AutoRoute(
    path: '/feed',
    page: FeedRoute.page,
  ),
];
```

The app is unable to resolve `/link/feed`, even though the deep link setup itself works on the native Android side. It results in the above-mentioned error.

**Temporary Workaround**

Currently, I am using the app_links package with a workaround like this:

```
deepLinkTransformer: (uri) {
  return Uri.parse('https://mydomain.com/login');
}
```

This “redirects” the deep link to the login route, where the logic to navigate to `/feed` if the user is already logged in is handled. While this works, it feels like a hack.

**Other question: what exactly does flutter_deeplinking_enabled do?**

This flag defaults to true from Flutter 3.27 and above, so I set it back explicitly to false. My assumption was that setting it to false would completely disable deep linking at the Router level, allowing the use of packages like app_links or uni_links. However, even with it set to false, auto_route still seems to interfere (at least when opening app from terminated state).

Environment
	•	Flutter: 3.27
	•	auto_route: 9.2.2

**Comment by AzarouAmine** [comment]
I think this is exactly how it is supposed to work. Your app does not have a page with a path "link", this is why the deep link builder does not recognize or does not know what to do with it.
